### PR TITLE
[ui-components ] Reskin of Inputs and Buttons to Match the Updated Designs

### LINKED ISSUE #61: Update UI

### PR DESCRIPTION
**Pull request by bravomartin**
This is a pretty significant change to the `mechanic-ui-components` and the app framework in `mechanic`.

- updates all ui components to use the new visual language. they can now be color-customized by setting a css `color` in their parent.
- updates the app framework to use the new visual language.

fixes #61

### CODE REVIEW

**Comment by bravomartin** [comment]
@fdoflorenzano this is ready to review.

**Review by fdoflorenzano — CHANGES_REQUESTED** [review]
Found some issues:

Image preview goes to back of other elements (Firefox and Chrome at least)
Screen Shot 2021-08-18 at 10 16 02

When opening color picker, another scroll appears (Firefox and Chrome) 
Screen Shot 2021-08-18 at 10 17 12

Blurry filter gets stuck in a certain position
Screen Shot 2021-08-18 at 10 19 40

I get this error when closing color picker (Firefox and Chrome) maybe its a yalc thing:
`Uncaught DOMException: Element.closest: '.-yalc-@designsystemsinternational-mechanic-ui-components-src-input---ColorInput-module__root .-yalc-@designsystemsinternational-mechanic-ui-components-src---common-module__input-wrapper' is not a valid selector`

For some reason, when testing the `dsi-logo-maker` in Firefox, I get a much bigger sidebar. This doesn't happen in Chrome or Safari. 
Screen Shot 2021-08-18 at 10 24 40
Screen Shot 2021-08-18 at 10 24 31

On Firefox I get all white select options
Screen Shot 2021-08-18 at 10 27 22

File: packages/mechanic/app/pages/Nav.js [hunk 1]
```diff
@@ -1,31 +1,46 @@
...
 import { useHistory } from "react-router-dom";
+import { DropdownIcon } from "@designsystemsinternational/mechanic-ui-components";
 import * as css from "./Nav.module.css";
 
-export const Nav = ({ name, functionsNames }) => {
+export const Nav = ({ name, functions }) => {
   let history = useHistory();
-
+  const functionsNames = Object.keys(functions);
+  console.log(functions);
```
Comment (fdoflorenzano) [inline packages/mechanic/app/pages/Nav.js:10]: Please remove.

File: packages/mechanic/app/pages/Nav.js [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
+  const currentFn = functions[name];
+  const currentLabel = currentFn.settings.name || Case.title(name);
```
Comment (fdoflorenzano) [inline packages/mechanic/app/pages/Nav.js:12]: There is a `name` setting also now? What are the rules of this new setting?

**Review by fdoflorenzano — COMMENTED** [review]
Lots of great work! A lot of removals 😄 

Also, CHANGELOGS are missing!

**Comment by bravomartin** [comment]
- [x] Image preview goes to back of other elements (Firefox and Chrome at least)

- [ ] When opening color picker, another scroll appears (Firefox and Chrome) — Can't reproduce

- [x] Blurry filter gets stuck in a certain position

- [ ] I get this error when closing color picker (Firefox and Chrome) maybe its a yalc thing. — it's because of the `@` in the path
- [ ] For some reason, when testing the `dsi-logo-maker` in Firefox, I get a much bigger sidebar. This doesn't happen in Chrome or Safari. 

- [x] On Firefox I get all white select options

File: packages/mechanic-fonts/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      "name": "Fernando Florenzano Hernández",
+      "email": "i@designsystems.international",
+      "url": "https://designsystems.international"
+    },
+    {
+      "name": "Martin Bravo",
+      "email": "i@designsystems.international",
+      "url": "https://designsystems.international"
+    }
+  ],
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-fonts/package.json:32]: : )

File: packages/mechanic-fonts/fonts.css [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  font-weight: bold;
+  font-style: italic;
+}
+
+@font-face {
+  font-family: Plex;
+  src: url("./IBM-Plex-Mono/IBMPlexMono-Regular.otf") format("opentype");
+  font-weight: normal;
+  font-style: normal;
+}
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-fonts/fonts.css:30]: I'll add a couple more. The files are only bundled if you require them, and only downloaded in the browser if actually used. I think it's ok to keep it like this and clean up once we have a clear idea of the variants we need.

File: packages/mechanic-fonts/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "style": "fonts.css"
+}
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-fonts/package.json:34]: will do

File: packages/mechanic-ui-components/app/index.js [hunk 6]
```diff
@@ -142,6 +143,20 @@ const App = () => {
...
+          attributes={{
+            type: "text",
+            editable: false,
+            default: "Option 1",
+            options: ["Option 1", "Option 2", "Option 3"],
+            validation: value => (value === "Option 1" ? null : "Should be Option 1")
+          }}
+          onChange={handleChange}
+        />
+      </div>
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-ui-components/app/index.js:159]: added before scrolling to see the following section. deleted.

File: packages/mechanic-ui-components/src/ParamInput.js [hunk 7]
```diff
@@ -10,10 +11,19 @@ import { ColorInput } from "./input/ColorInput.js";
...
+  attributes,
+  onChange,
+  children,
+  color = "#f04b17"
+}) => {
   const id = useRef(uid("param-input"));
-  const { type, options, validation, editable } = attributes;
+  const { type, label: _label, options, validation, editable } = attributes;
   const _default = attributes["default"];
+  const label = Case.title(name) || _label;
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-ui-components/src/ParamInput.js]: fixed

File: packages/mechanic-ui-components/src/ParamInput.js [hunk 8]
```diff
@@ -130,10 +140,9 @@ export const ParamInput = ({ name, className, values, attributes, onChange, chil
     <TextInput
```
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-ui-components/src/ParamInput.js:127]: removed color prop altogether

File: packages/mechanic/app/pages/Nav.js (hunk 2 above)
Comment (bravomartin) [reply packages/mechanic/app/pages/Nav.js:12]: adds the option to name the function some thing different than the key. designers will appreciate this

File: packages/mechanic/app/pages/Nav.js (hunk 1 above)
Comment (bravomartin) [reply packages/mechanic/app/pages/Nav.js]: removed

File: packages/mechanic-ui-components/src/ParamInput.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-ui-components/src/ParamInput.js]: flipped the order and it now makes sense. 
also added validation

File: packages/mechanic-fonts/package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-fonts/package.json:45]: and some more plex variants and a  files array

File: packages/mechanic-fonts/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (bravomartin) [reply packages/mechanic-fonts/package.json]: spotted. ok its me.

**Comment by bravomartin** [comment]
@fdoflorenzano please take another look

File: packages/mechanic-ui-components/src/input/NumberInput.module.css [hunk 9]
```diff
@@ -24,53 +10,72 @@
...
 
 /* https://css-tricks.com/styling-cross-browser-compatible-range-inputs-css/ */
 .range-input {
+  position: relative;
+  z-index: 9;
   flex: 1 1 auto;
   min-width: 0;
-
+  color: inherit;
   background: transparent; /* Otherwise white in Chrome */
```
Comment (fdoflorenzano) [inline packages/mechanic-ui-components/src/input/NumberInput.module.css:64]: I dug dipper into this mysterious bug that made some test have a bigger sidebar in Firefox. I realized it was the range number input specifically, design functions that had a range number param had a much bigger sidebar only on Firefox:

Screen Shot 2021-08-23 at 10 50 20
Screen Shot 2021-08-23 at 10 50 25

I did some googling and found [here](https://brennaobrien.com/blog/2014/05/style-input-type-range-in-every-browser.html) that for Firefox: "You'll also need to give input[type=range] the same width as your track. It won't expand to contain the track on it's own." I locally added here `width: 100%;` and it stopped doing the weird resizing. I'll let you decide if that makes sense here. Hope this helps!

**Review by fdoflorenzano — COMMENTED** [review]
It's looking even much better!! There is only the + symbol issue we saw during our call and the mysterious big sidebar thing for Firefox. I left a comment about it with a potential fix.

**Comment by bravomartin** [comment]
done! please let me know if you find anything else @fdoflorenzano

**Review by fdoflorenzano — APPROVED** [review]
🥇

**Pull request merged**